### PR TITLE
feat: create playlists from errors

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We want users to be able to see a playlist of recordings containing a certain error

## Changes

- Create playlist
- Pin recordings individually. Note: This is temporary while I work on a refactor to allow the playlist UI render without filters / session loading code (in the works but complicated)

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

https://github.com/PostHog/posthog/assets/6685876/fa3acc4f-6551-4394-8edc-44e505b28e72

### CODE REVIEW

File: ee/session_recordings/ai/error_clustering.py [hunk 1]
```diff
@@ -76,12 +76,12 @@ def construct_response(df: pd.DataFrame, team: Team, user: User):
     clusters = []
     for cluster, rows in df.groupby("cluster"):
         session_ids = rows["session_id"].unique()
-        sample = rows.sample(n=1)[["session_id", "input"]].rename(columns={"input": "error"}).to_dict("records")
+        sample = rows.sample(n=1)[["session_id", "input"]].rename(columns={"input": "error"}).to_dict("records")[0]
```
Comment (pauldambra) [inline ee/session_recordings/ai/error_clustering.py:79]: silly check do we know we have a non-empty thing to get `[0]` from at this point?

File: frontend/src/scenes/session-recordings/errors/sessionRecordingErrorsLogic.ts [hunk 2]
```diff
@@ -19,7 +25,18 @@ export const sessionRecordingErrorsLogic = kea<sessionRecordingErrorsLogicType>(
             },
         ],
     })),
+    listeners(() => ({
+        createPlaylist: async ({ name, sessionIds }) => {
+            const playlist = await createPlaylist({ name: name })
 
+            if (playlist) {
+                await Promise.all(
+                    sessionIds.map((sessionId) => api.recordings.addRecordingToPlaylist(playlist.short_id, sessionId))
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/errors/sessionRecordingErrorsLogic.ts:34]: 🤔 we should maybe limit the size of the recordings we will add to the playlist here

e.g. if there are 4500 items in a cluster do we want to add them all to a playlist

and do we want to make 4.5k API calls?

i'd definitely do at least one of limiting the list or amending the API to take a list of ids

**Review by pauldambra — CHANGES_REQUESTED** [review]
one comment on the playlist creation...

File: frontend/src/scenes/session-recordings/errors/sessionRecordingErrorsLogic.ts (hunk 2 above)
Comment (marandaneto) [reply frontend/src/scenes/session-recordings/errors/sessionRecordingErrorsLogic.ts:34]: would be possible to paginate? so we query the next 100s if the user goes to the next 100

File: ee/session_recordings/ai/error_clustering.py (hunk 1 above)
Comment (daibhin) [reply ee/session_recordings/ai/error_clustering.py:79]: Yep good question. The cluster will only be generated if it meets the `MIN_SAMPLES` so we know that each cluster has _at least_ 10 rows

File: frontend/src/scenes/session-recordings/errors/sessionRecordingErrorsLogic.ts (hunk 2 above)
Comment (daibhin) [reply frontend/src/scenes/session-recordings/errors/sessionRecordingErrorsLogic.ts]: Good call. In this instance we were protected by the fact that only 10 samples are returned from the backend: https://github.com/PostHog/posthog/pull/21037/files#diff-18f37520b0e5b0957994182f1c1b6b582893c2d04e87b88e75e967cd6fcc64abR84. But no harm slicing the array on the frontend too. Pushed a new commit

---

@marandaneto think that is the way to go once we have split out static / dynamic playlists (next on my list but the refactor is taking longer than expected)

**Review by pauldambra — APPROVED** [review]
:shipit:

**Review by marandaneto — APPROVED** [review]

**Pull request merged**